sound/soc/sof/sof-audio.c

```diff
@@ -53,6 +53,52 @@ int sof_set_hw_params_upon_resume(struct device *dev)
 	return snd_sof_dsp_hw_params_upon_resume(sdev);
 }
 
+int sof_destroy_pipelines(struct device *dev)
```

Comment:
i am still confused why this needs to be done by hand and isn't done by topology and friends?

Reply:
Good idea. Let me move it to topology.

Reply:
@plbossart sorry I dont think I can be done by topology. When removing the SOF device, topology can handle this but when dealing with runtime PM, it has be done here.